server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/documents.ts

```diff
@@ -162,9 +163,13 @@ export function create(
 				? uuid()
 				: (request.body.id as string) || uuid();
 
+			Lumberjack.info(
+				`Whole summary length= ${JSON.stringify(request.body.summary).length}.`,
+			);
 			// Summary information
-			const summary = request.body.summary;
+			const summary = convertWholeSummaryTreeToSummaryTree(request.body.summary);
```

Comment:
If I got you right, you are changing protocol between client & service. There needs to be some consideration for back compatibility. I.e. old client talking to new server, and new client talking to old server. We can eliminate one pair by timing properly release cadence (i.e. if we ship service changes first and wait for it to propagate through the system, then new client - old service pair is eliminated), but there is no way to eliminate both pairs.
Given that clients take > 3 years to move off old bits, I'd suggest adding auto-detection mechanism in service code to work with both formats.

Q: Does our tests capture that need? If not, feels like we need to remedy this somehow.

Reply:
Great point! I can easily add a flag to the protocol, indicating the new format and only convert if the flag is sent.